pkg/simple/client/devops/fake/fakedevops.go

```diff
@@ -0,0 +1,120 @@
+func (d *FakeDevops)GetNodeSteps(projectName, pipelineName, runId, nodeId string, httpParameters *devops.HttpParameters) ([]devops.NodeSteps, error){
+	res := []devops.NodeSteps{
+		{ID:nodeId,Result:"true",DisplayName:"fakeNodeStep"},
+	}
+
+	return res, nil
+}
+func (d *FakeDevops)GetPipelineRunNodes(projectName, pipelineName, runId string, httpParameters *devops.HttpParameters) ([]devops.PipelineRunNodes, error){
+	res := []devops.PipelineRunNodes{
+		{ID:"1",Result:"true",DisplayName:"fakePipelineRunNode1"},
+		{ID:"2",Result:"true",DisplayName:"fakePipelineRunNode2"},
+		{ID:"3",Result:"true",DisplayName:"fakePipelineRunNode3"},
+	}
+
+	return res, nil
+}
+func (d *FakeDevops)SubmitInputStep(projectName, pipelineName, runId, nodeId, stepId string, httpParameters *devops.HttpParameters) ([]byte, error){return nil,nil}
+
+//BranchPipelinne operator interface
+func (d *FakeDevops)GetBranchPipeline(projectName, pipelineName, branchName string, httpParameters *devops.HttpParameters) (*devops.BranchPipeline, error){return nil,nil}
+func (d *FakeDevops)GetBranchPipelineRun(projectName, pipelineName, branchName, runId string, httpParameters *devops.HttpParameters) (*devops.PipelineRun, error){return nil,nil}
+func (d *FakeDevops)StopBranchPipeline(projectName, pipelineName, branchName, runId string, httpParameters *devops.HttpParameters) (*devops.StopPipeline, error){return nil,nil}
+func (d *FakeDevops)ReplayBranchPipeline(projectName, pipelineName, branchName, runId string, httpParameters *devops.HttpParameters) (*devops.ReplayPipeline, error){return nil,nil}
+func (d *FakeDevops)RunBranchPipeline(projectName, pipelineName, branchName string, httpParameters *devops.HttpParameters) (*devops.RunPipeline, error){return nil,nil}
+func (d *FakeDevops)GetBranchArtifacts(projectName, pipelineName, branchName, runId string, httpParameters *devops.HttpParameters) ([]devops.Artifacts, error){return nil,nil}
+func (d *FakeDevops)GetBranchRunLog(projectName, pipelineName, branchName, runId string, httpParameters *devops.HttpParameters) ([]byte, error){return nil,nil}
+func (d *FakeDevops)GetBranchStepLog(projectName, pipelineName, branchName, runId, nodeId, stepId string, httpParameters *devops.HttpParameters) ([]byte, http.Header, error){return nil,nil,nil}
+func (d *FakeDevops)GetBranchNodeSteps(projectName, pipelineName, branchName, runId, nodeId string, httpParameters *devops.HttpParameters) ([]devops.NodeSteps, error){
+	res := []devops.NodeSteps{
+		{ID:nodeId,Result:"true",DisplayName:"fakeBranchNodeStep"},
+	}
+	return res,nil
+}
+func (d *FakeDevops)GetBranchPipelineRunNodes(projectName, pipelineName, branchName, runId string, httpParameters *devops.HttpParameters) ([]devops.BranchPipelineRunNodes, error){
+	res := []devops.BranchPipelineRunNodes{
+		{ID:"1",Result:"true",DisplayName:"fakeBranchPipelineRunNode1"},
```

Comment:
It is recommended to add some fake data when fakeClient is initialized.
like kubernetes fake client 
https://github.com/kubesphere/kubesphere/pull/1739/files#diff-97e1af562039b7f55eb13ab7660d9a0eR41